### PR TITLE
prg - mantis_28028 - add confirmation for delete categories

### PR DESCRIPTION
**Pull request by daniwe4**
prg + crs survey exercise session - mantis_27976 - adjust lng vars for mail template gui

### CODE REVIEW

File: Modules/StudyProgramme/classes/class.ilObjStudyProgrammeAutoMembershipsGUI.php [hunk 1]
```diff
@@ -238,32 +264,98 @@ protected function save()
...
-            }
+
+        if (!array_key_exists($field, $post)) {
+            ilUtil::sendInfo($this->lng->txt('prg_delete_nothing_selected'), true);
+            $this->ctrl->redirect($this, self::CMD_VIEW);
+        }
+
+        $str = '';
+        foreach ($post[$field] as $data) {
+            $str .= $data . ' ';
```
Comment (klees) [inline Modules/StudyProgramme/classes/class.ilObjStudyProgrammeAutoMembershipsGUI.php]: Use [`implode`](https://www.php.net/manual/en/function.implode.php) instead.

File: Modules/StudyProgramme/classes/class.ilObjStudyProgrammeAutoMembershipsGUI.php [hunk 2]
```diff
@@ -238,32 +264,98 @@ protected function save()
...
+        ];
+
+        $message_box = $this->message_box_factory->confirmation($message)->withButtons($buttons);
+
+        $this->tpl->setContent($this->ui_renderer->render($message_box));
+    }
+
+    protected function delete()
+    {
+        $get = $_GET;
```
Comment (klees) [inline Modules/StudyProgramme/classes/class.ilObjStudyProgrammeAutoMembershipsGUI.php]: We already have the request. Let's use it here.

File: Modules/StudyProgramme/classes/class.ilObjStudyProgrammeAutoMembershipsGUI.php [hunk 3]
```diff
@@ -238,32 +264,98 @@ protected function save()
...
+            $this->getObject()->deleteAutomaticMembershipSource((string) $type, (int) $id);
+        }
+
+        ilUtil::sendSuccess($this->lng->txt('prg_delete_success'), true);
+        $this->ctrl->redirect($this, self::CMD_VIEW);
+    }
+
+    protected function deleteSingleConfirmation()
+    {
+        $get = $_GET;
```
Comment (klees) [inline Modules/StudyProgramme/classes/class.ilObjStudyProgrammeAutoMembershipsGUI.php]: Use `$this->request` here as well.

File: Modules/StudyProgramme/classes/class.ilObjStudyProgrammeAutoMembershipsGUI.php [hunk 4]
```diff
@@ -238,32 +264,98 @@ protected function save()
...
+        $delete = $this->ctrl->getFormActionByClass(self::class, self::CMD_DELETE_SINGLE);
+        $cancel = $this->ctrl->getFormActionByClass(self::class, self::CMD_VIEW);
+        $this->ctrl->clearParameterByClass(self::class, $field);
+
+        $buttons = [
+            $this->button_factory->standard($this->lng->txt('prg_confirm_delete'), $delete),
+            $this->button_factory->standard($this->lng->txt('prg_cancel'), $cancel)
+        ];
+
+        $message_box = $this->message_box_factory->confirmation($message)->withButtons($buttons);
```
Comment (klees) [inline Modules/StudyProgramme/classes/class.ilObjStudyProgrammeAutoMembershipsGUI.php]: This code seems to be mostly duplicated with `deleteConfirmation`. Can we unify it?

**Review by klees — CHANGES_REQUESTED** [review]
Hi Daniel,

LGTM mostly, please try to deduplicate the code and use the request object instead of the superglobals to access GET-parameters.

Thanks!

File: Modules/StudyProgramme/classes/class.ilObjStudyProgrammeAutoMembershipsGUI.php (hunk 1 above)
Comment (daniwe4) [reply Modules/StudyProgramme/classes/class.ilObjStudyProgrammeAutoMembershipsGUI.php]: done

File: Modules/StudyProgramme/classes/class.ilObjStudyProgrammeAutoMembershipsGUI.php (hunk 2 above)
Comment (daniwe4) [reply Modules/StudyProgramme/classes/class.ilObjStudyProgrammeAutoMembershipsGUI.php]: done

File: Modules/StudyProgramme/classes/class.ilObjStudyProgrammeAutoMembershipsGUI.php (hunk 3 above)
Comment (daniwe4) [reply Modules/StudyProgramme/classes/class.ilObjStudyProgrammeAutoMembershipsGUI.php]: done

File: Modules/StudyProgramme/classes/class.ilObjStudyProgrammeAutoMembershipsGUI.php (hunk 4 above)
Comment (daniwe4) [reply Modules/StudyProgramme/classes/class.ilObjStudyProgrammeAutoMembershipsGUI.php]: done

**Comment by klees** [comment]
I merged this manually.

**Pull request closed without merging**